File: lib-segment/src/main/java/websiteschema/mpsegment/core/IConceptRecognizer.java
package websiteschema.mpsegment.core;

import websiteschema.mpsegment.dict.IWord;

public interface IConceptRecognizer {

    public void setWordArray(IWord[] wordArray);

    public void setPosArray(int[] posArray);

    public String[] getConcepts();

    public void reset();
}

File: lib-segment-scala/src/main/scala/websiteschema/mpsegment/core/IConceptRecognizer.scala
package websiteschema.mpsegment.core

import websiteschema.mpsegment.dict.IWord

trait IConceptRecognizer {

    def setWordArray(wordArray: Array[IWord])

    def setPosArray(posArray: Array[Int])

    def getConcepts() : Array[String]

    def reset()
}
